.github/workflows/main.yml

```diff
@@ -38,6 +38,50 @@ jobs:
     - run: rustup update --no-self-update stable && rustup default stable
     - run: cargo check --all
 
+  # Run `cargo clippy` over everything
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: >-
+        -D warnings
+        -A dead_code
+        -A non_upper_case_globals
+        -A unused_doc_comments
+        -A clippy::disallowed_names
+        -A clippy::enum_variant_names
+        -A clippy::large_enum_variant
+        -A clippy::missing_safety_doc
+        -A clippy::new_without_default
+        -A clippy::overly_complex_bool_expr
+        -A clippy::too_many_arguments
+        -A clippy::type_complexity
```

Comment:
I think it would be really important to put these lints into the code somehow, otherwise contributors won't notice an issue until the make a PR and it fails on CI.

We could add them to a global `.cargo/config.toml`?

Reply:
I agree. The problem with `.cargo/config.toml` is that the file shouldn't be checked in to git, like [the comment](https://github.com/rustwasm/wasm-bindgen/blob/0.2.84/.cargo/config#L1-L2) suggests. I'll try to move these exceptions into the crates that need it (shouldn't be that much duplication).

Reply:
I'm not sure why it was decided that this shouldn't be checked into git, it's common solution, the Cargo documentation also doesn't indicate that one shouldn't check this into git either.

Reply:
Ok, I have added .config/cargo with allow rustflags. Warnings should not be generally denied, so the CI still needs to set RUSTFLAGS manually, which overrides .cargo/config, so the allows are duplicated once. I added comments in both directions, so it should be acceptable.